hardware/esp8266com/esp8266/libraries/ESP8266HTTPUpdateServer/src/ESP8266HTTPUpdateServer.cpp

```diff
@@ -0,0 +1,88 @@
+#include <Arduino.h>
+#include <WifiClient.h>
+#include <WifiServer.h>
+#include <ESP8266WebServer.h>
+#include <WifiUDP.h>
+#include "ESP8266HTTPUpdateServer.h"
+
+
+const char* ESP8266HTTPUpdateServer::_serverIndex = "<html><body><form method='POST' action='/update' enctype='multipart/form-data'>"
+                                          "<input type='file' name='update'>"
+                                          "<input type='submit' value='Update'>"
+                                      "</form>"
+                                "</body></html>";
+
+ESP8266HTTPUpdateServer::ESP8266HTTPUpdateServer(int port, bool serial_debug)
+{
+  _serial_output = serial_debug;
+  _server = NULL;
+  _port = port;
+}
+
+ESP8266HTTPUpdateServer::~ESP8266HTTPUpdateServer()
+{
+  if (_server)
+    delete _server;
```

Comment:
This doesn't look right in case when ESP8266WebServer instance was allocated statically outside this class. Consider the following code:
```c++
void runServer() {
  ESP8266WebServer httpServer(80);
  ESP8266HTTPUpdateServer updateServer;
  updateServer.setup(&httpServer);
  while (something) {
    httpServer.handleClient();
  }
}
```
Upon exit from this function, ESP8266WebServer will be deleted twice.